In [1]:
import csv
import pandas as pd
import numpy as np
#from sklearn import
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import OneHotEncoder
from sklearn.metrics import mean_squared_error
import matplotlib.pyplot as plt
import statsmodels.api as sm
from scipy import stats
%matplotlib inline

## Years to Load

In [22]:
years = [14, 15]

In [23]:
taxi_df = {}

for i in years:
    taxi_df[i] = pd.read_csv("../dataset_4_02_modified/20" + str(i) \
                            + "lag.csv", sep = ",")
# "2016" and before
# taxi_df[15] = pd.concat([taxi_df[15], taxi_df[16]], axis = 0)\
#               .reset_index().drop(columns = ['index'])

# Naive Lookup Previous Year

In [24]:
for i in years:
    taxi_df[i]["Short Timestamp"] = pd.to_datetime(taxi_df[i]["Trip Start Timestamp"], format='%Y/%m/%d %H')\
                                    .dt.strftime('%m/%d %H')

In [25]:
lookup_df = taxi_df[14][["Short Timestamp", "Count", "Pickup Community Area"]]\
            .merge(right = taxi_df[15][["Short Timestamp", "Count", "Pickup Community Area"]],\
                   on = ["Short Timestamp", "Pickup Community Area"], how = "inner")

## Mean Square Error

In [26]:
mean_squared_error(y_true = lookup_df["Count_x"], y_pred =  lookup_df["Count_y"])

5057.89004051186

# Naive Lookup Previous Hour

## Mean Square Error

In [21]:
mean_squared_error(y_true = taxi_df[15]["Count"], y_pred = taxi_df[15]["lag1"])

941.1593802531947

### Results
#### Naive Lookup Previous Year: 2015 to 2016 MSE = 6557
#### Naive Lookup Previous Hour: 2016 MSE = 607


#### Naive Lookup Previous Year: 2015 to 2016 MSE = 5057
#### Naive Lookup Previous Hour: 2015 MSE = 941